### PR TITLE
docs(bpmn-workflows): clarify variable visibility of multi-instance

### LINKED ISSUE #4362: Describe the visibility and propagation of variables within a multi-instance activity
**Description**
Enrich the [multi-instance documentation](https://docs.zeebe.io/bpmn-workflows/multi-instance/multi-instance.html) by a description of
* which variables are visible within an inner instance
* how to limit the variables to an inner instance

> We would expect that the instances are scoped and can only access root-workflow variables, variables scoped to the subprocess and local variables.

> If the variable "record" should not be visible outside of the multi-instance subprocess then it must be declared by an input variable mapping.

### PR DESCRIPTION
**Pull request by saig0**
## Description

* extend variable mapping section of multi-instance page

## Related issues

closes #4362

## Pull Request Checklist

- [x] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [ ] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [ ] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

File: docs/src/bpmn-workflows/multi-instance/multi-instance.md [hunk 1]
```diff
@@ -65,6 +65,8 @@ Every instance has a local variable `loopCounter`. It holds the index in the `in
 
 Input and output variable mappings can be defined at the multi-instance activity. They are applied **on each instance** on activating and on completing.
 
+The input mappings can be used to create new local variables in the scope of an instance. These variables are only visible within the instance. It is a way to restrict the **visibility** of variables. By default, new variables, for example, provided by a job worker, are created in the scope of the workflow instance and are visible to all instances and outside of the multi-instance activity. If the variable is defined as local variable then it is not propagated to a parent or the workflow instance scope.
```
Comment (korthout) [inline docs/src/bpmn-workflows/multi-instance/multi-instance.md:68]:
```suggestion
The input mappings can be used to create new local variables in the scope of an instance. These variables are only visible within the instance. It is a way to restrict the **visibility** of variables. By default, new variables (e.g. provided by a job worker) are created in the scope of the workflow instance and are visible to all instances (i.e. instances of the multi instance activity) as well as outside of the multi-instance activity. If the variable is defined as local variable, then it is not propagated to a parent or the workflow instance scope. This is especially important for parallel multi instance activities, as race conditions may occur when variables are shared among the instances (i.e. non-local variables).
```

**Review by korthout — CHANGES_REQUESTED** [review]
It's a good addition, but I would add something about potential race conditions when using non-local variables in parallel multi instance activities. (At least, I think there might be, so correct me if I'm wrong).

**Comment by saig0** [comment]
That is a good hint :+1:  Please have a second look.

**Review by korthout — APPROVED** [review]
LGTM

**Pull request merged**